src/clj/game/cards.clj

```diff
@@ -207,6 +207,42 @@
                                                (vals selected-cards)))]
                            (effect-completed state side (make-result eid {:number counter-count :msg msg})))))}))
 
+(defn pick-credit-providing-cards
+  "Similar to pick-virus-counters-to-spend. Works on :recurring and normal credits."
+  ([provider-func] (pick-credit-providing-cards (hash-map) 0 nil provider-func))
+  ([target-count provider-func] (pick-credit-providing-cards (hash-map) 0 target-count provider-func))
+  ([selected-cards counter-count target-count provider-func]
+   {:async true
+    :prompt (str "Select a card with recurring credits ("
+                 counter-count (when (and target-count (pos? target-count))
+                                 (str " of " target-count))
+                 " credits)")
+    :choices {:req #(and (< 0 (get-counters % :recurring)) (.contains (provider-func) %))}
```

Comment:
Could use `pos?` here. And I feel like `.contains` is not quite what we want to use, but I'll have to play with it to see if there's a better method.

Reply:
`pos?` seems to be a function to check for positive numbers.. Sure you mean that one?

I wanted to use `contains?` but it seems to only work for keys of maps.

Reply:
I have added `in-coll?` in `utils.clj` to do this.